app/src/main/java/org/fossasia/openevent/general/discount/DiscountCode.kt

```diff
@@ -7,25 +7,27 @@ import com.github.jasminb.jsonapi.annotations.Id
 import com.github.jasminb.jsonapi.annotations.Relationship
 import com.github.jasminb.jsonapi.annotations.Type
 import org.fossasia.openevent.general.event.Event
+import org.fossasia.openevent.general.ticket.Ticket
 
 @Type("discount-code")
 @JsonNaming(PropertyNamingStrategy.KebabCaseStrategy::class)
 data class DiscountCode(
     @Id(IntegerIdHandler::class)
     val id: Int,
     val code: String,
-    val discountUrl: String? = null,
-    val type: String,
-    val value: Float,
-    val ticketsNumber: Int?,
-    val usedFor: String,
-    val tickets: String? = null,
-    val maxQuantity: Int?,
-    val minQuantity: Int?,
-    val isActive: Boolean = false,
     val validFrom: String? = null,
-    val validTill: String? = null,
+    val minQuantity: Int? = null,
     val createdAt: String? = null,
+    val ticketsNumber: Int? = null,
+    val value: Float? = null,
+    val maxQuantity: Int? = null,
+    val isActive: Boolean = false,
+    val usedFor: String,
+    val validTill: String? = null,
+    val discountUrl: String? = null,
+    val type: String,
     @Relationship("event")
-    val event: Event
+    val event: Event? = null,
+    @Relationship("tickets")
+    val tickets: List<Ticket>? = null
```

Comment:
I'm getting following response - 
```
DiscountCode(id=7, code=FOSSASIA, validFrom=2019-06-10T05:48:37.962000+00:00, minQuantity=1, createdAt=2019-06-10T05:49:06.179000+00:00, ticketsNumber=5, value=10.0, maxQuantity=1, isActive=true, usedFor=ticket, validTill=2019-08-16T18:30:00+00:00, discountUrl=https://fossasia.github.io/open-event-frontend/e/7376e026?code=FOSSASIA, type=amount, event=null, tickets=null)
```

How can I store that relationship object to relate it with event and tickets - 
```
"relationships": {
      "events": {
        "links": {
          "self": "/v1/discount-codes/4/relationships/events",
          "related": "/v1/discount-codes/4/events"
        }
      },
      "tickets": {
        "links": {
          "self": "/v1/discount-codes/1/relationships/tickets",
          "related": "/v1/discount-codes/1/tickets"
        }
      }
    }
```

For this DiscountCode class, they are getting null.

Reply:
They are null. You haven't included them in the query. Secondly, instead of getting tickets of discount code, you should get discount code of ticket

Reply:
> you should get discount code of ticket

I thought it will also give `HTTP 401 UNAUTHORIZED`. Working :+1:

Reply:
> HTTP 401 UNAUTHORIZED

Same for end point: `/v1/tickets/{ticket_id}/discount-codes{?sort,filter}`

Reply:
Yeah, I remebered. Obviously you can't see the discount codes. Or what would be the point. Imagine DigitalOcean API showing that BLACKOUT100 will give you 100% off 

Why you need to fetch the discount codes?

Reply:
Then what would be the solution, Which end-point should I use first?

Reply:
> Imagine DigitalOcean API showing that BLACKOUT100 will give you 100% off
> 
> Why you need to fetch the discount codes?

We also need to check if discount code is active, what is the time limit and min & max Qty in that case.

Reply:
Coordinate with server devs

Reply:
> Coordinate with server devs

Sorry, but for what? It is working fine!

Reply:
HTTP 401 UNAUTHORIZED

Same for end point: /v1/tickets/{ticket_id}/discount-codes{?sort,filter}

Reply:
But is it okay to fetch these discount codes of tickets for a non-admin user?!

Reply:
No, it is not. So, how are you implementing it?

Reply:
Okay, got it. You want to say that endpoint `discount-codes/{code}` should also give `HTTP 401 UNAUTHORIZED`?

Reply:
Which I'm using currently

Reply:
OK, if you have got a discount code, it doesn't throw the error? If so, then it is fine :+1:

Reply:
> OK, if you have got a discount code, it doesn't throw the error? If so, then it is fine

Yes, that's I'm using. Fetching details after submitting the discount code.

Reply:
> If so, then it is fine

@iamareebjamal Please review!